### PR TITLE
Passwordless Auth - Adjust font and spacing 

### PR DESCRIPTION
**Pull request by moon0326**
This PR updates the CSS styles for the passwordless auth pages.

## Proposed Changes

- Base font is `SF Pro Text`
- H1 and H3 headers use `SF Pro Display`
- Fixed styling issues with the social buttons due to changes from the trunk. The buttons should have `SF Pro Display` font family and centered.
- Spacing between headers and the subtitles should be 15px.

The SF Pro font family isn't available on Windows systems. I explored the option of using the web font version but am uncertain about its legal usage. I'm looking into it.

## Why are these changes being made?

* To reflect the Figma design

## Testing Instructions

1. Setup https://github.com/woocommerce/woocommerce-start-dev-env 
2. Once https://github.com/woocommerce/woocommerce-start-dev-env is ready, cd to wp-calypso
3. Open `config/development.json` and enable [woocommerce/core-profiler-passwordless-auth](https://github.com/Automattic/wp-calypso/blob/trunk/config/development.json#L250)
4. Run `yarn start`
5. Open a new incognito window (Make sure you're logged out from WPCOM)
6. Create a new JN site with the latest WooCommerce
7. Go through the core profiler and make sure to select Jetpack on the plugins page.
8. Once you get to the jetpack connect page, right click on `One last step!` header and inspect it.
9. Make sure the rendered font is `SF Pro Display`.
10. Right click on the subtitle and inspect it. Make sure the rendered font is `SF Pro Text`
11. Confirm the spacing between the header and the subtitle is 15px.
12. Click on the `Log in` link
13. Inspect the header and the subtitle and confirm the rendered font.
14. Confirm the social buttons are correctly centered and use `SF Pro Display` font.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/layout/masterbar/woo.scss [hunk 1]
```diff
@@ -1895,10 +1897,21 @@ $breakpoint-mobile: 660px;
 
 	}
 
+	// Core Profiler + Passwordless
 	&.is-woocommerce-core-profiler-flow.is-woo-passwordless {
```
Comment (psealock) [inline client/layout/masterbar/woo.scss:1901]: On the connection page I replaced `wordpress.com` with Calypso localhost but couldn't find the styles. I think because there is no `is-woo-passwordless` class applied? Have I landed on the correct page?

image

Comment (moon0326) [reply client/layout/masterbar/woo.scss:1901]: @psealock My bad 🙇 I forgot to add a step to enable the feature flag. I've updated the test instructions.

tl;dr -- you should enable https://github.com/Automattic/wp-calypso/blob/trunk/config/development.json#L250 feature flag.

**Review by chihsuan — APPROVED** [review]
Thanks, font-family changes look good to me! Only one small thing I noticed, the font-
size of primary button is 16px, but it's 14px in the design (Y5pUYSJPsGEud1vknUZhi8-fi-3182_14032). Should we update it to 14px? 

Screenshot 2024-09-18 at 16 07 26


Pre-approving!
